include/openthread/types.h

```diff
@@ -683,6 +683,8 @@ enum
     OT_CHANGED_THREAD_CHILD_REMOVED         = 1 << 11,  ///< Child was removed
     OT_CHANGED_IP6_MULTICAST_SUBSRCRIBED    = 1 << 12,  ///< Subscribed to a IPv6 multicast address
     OT_CHANGED_IP6_MULTICAST_UNSUBSRCRIBED  = 1 << 13,  ///< Unsubscribed from a IPv6 multicast address
+    OT_CPU_POWER_HI                         = 1 << 14,  ///< Hi CPU power is needed
```

Comment:
As I described earlier, i also do not think this is a correct thing.

It's not CPU_HIGH_POWER - that is a broad and blunt term.

The problem is 100% the crypto functions, and the computational horse power required to perform the operation.  

Option (A) is to use dedicated hardware, replacing the slow software method.

Option (B) is to use a compile time flag (ie: -Ofast) as I described

Option (C) is as suggested by this PR, "bump the cpu clock rate"

The problems I have here are:

a) the solution is describing a FIX, not the problem, or state the system is in.

b) The change should be limited to the time the CRYPTO activity is going on.

c) In this specific example, the problem arrives during JOINING state or COMMISSIONING.  What happens when the device needs to make (or accept) a DTLS connection from some *off*network application? And thus requires a higher CPU clock rate. Do we introduce another state change?

d) There is zero need for the CPU to be in "high clock speed" (and thus higher power) mode when it is waiting for packets (ie: BETWEEN crypto operations

That's why I would rather see CRYPTO code bracketed with something like OT_CRYTPO_BEGIN() and a OT_CRYPTO_END() macros. The macros can be defined by the platform to do as it needs.

Which in the DA15000 case is "clock high" and "clock normal"

Other platforms may for have solutions *OTHER* then  speed up the CPU clock, which is the solution for the DA15000 

For example:
    Maybe the BUS fabric between the CPU and the FLASH needs to be sped up
    Maybe a CPU opcode cache needs to be turned on (or off)
    Or the memory needs to be put into a different state so it responds faster.

None of these are "CPU_HIGH"